File: src/AbstractFactory/Models/Motorcycle.java
package AbstractFactory.Models;

import AbstractFactory.Factories.IFactory;
import AbstractFactory.Products.Bonnet.IBonnet;
import AbstractFactory.Products.Chain.IChain;
import AbstractFactory.Products.Chassis.IChassis;
import AbstractFactory.Products.Door.IDoor;
import AbstractFactory.Products.Handlebars.IHandlebars;
import AbstractFactory.Products.Motor.IMotor;
import AbstractFactory.Products.Seat.ISeat;
import AbstractFactory.Products.SteeringWheel.ISteeringWheel;
import AbstractFactory.Products.TailPipe.ITailPipe;
import AbstractFactory.Products.Wheel.IWheel;

public class Motorcycle implements IVehicle {
    private IWheel[] wheels = new IWheel[2];
    private IChassis chassis;
    private IMotor motor;
    private ISeat seat;
    private ITailPipe tailPipe;
    private IChain chain;
    private IHandlebars handlebars;

    public Motorcycle(IFactory factory){
        this.wheels[0] = factory.createWheel(14, 3);
        this.wheels[1] = factory.createWheel(14, 3);
        this.chassis = factory.createChassis();
        this.motor = factory.createMotor();
        this.seat = factory.createSeat("negro", "30x25");
        this.tailPipe = factory.createTailPipe();
        this.chain = factory.createChain();
        this.handlebars = factory.createHandlebars();
    }

    public String getVehicleInformation(){
        return String.format("" +
            "La moto cuenta con %d ruedas caracterizadas por: %s. %n" +
            "%s. %n" +
            "Motor %s. %n" +
            "%s. %n" +
            "%s. %n" +
            "%s. %n" +
            "%s.%n",
            this.wheels.length, this.wheels[0].getInformation(),
            this.chassis.getInformation(),
            this.motor.getInformation(),
            this.seat.getInformation(),
            this.tailPipe.getInformation(),
            this.chain.getInformation(),
            this.handlebars.getInformation());
    }
}

File: src/Builder/Models/Motorcycle.cs
using Builder.Products;
using System.Collections.Generic;
using System;

namespace Builder.Models{
    public class Motorcycle : IVehicle{
        public List<IProduct> wheels {get; set;}
        public List<IProduct> doors {get; set;}
        public IProduct bonnet {get; set;}
        public IProduct chassis {get; set;}
        public IProduct motor {get; set;}
        public List<IProduct> seats {get; set;}
        public IProduct steeringWheel {get; set;}
        public IProduct tailPipe {get; set;}
        public IProduct frame {get; set;}
        public IProduct handlebars {get; set;}
        public IProduct chain {get; set;}

        public Motorcycle(){
            this.wheels = new List<IProduct>();
            this.seats = new List<IProduct>();
        }

        public String getVehicleInformation(){
            return String.Format("" +
                "La moto cuenta con {0} ruedas caracterizadas por: {1}. " + Environment.NewLine +
                "{2} ." + Environment.NewLine +
                "Motor {3}." + Environment.NewLine +
                "{4} ." + Environment.NewLine +
                "{5} ." + Environment.NewLine +
                "{6} ." + Environment.NewLine +
                "{7}." + Environment.NewLine,
                this.wheels.Count, this.wheels[0].getInformation(),
                this.chassis.getInformation(),
                this.motor.getInformation(),
                this.seats[0].getInformation(),
                this.tailPipe.getInformation(),
                this.chain.getInformation(),
                this.handlebars.getInformation());
        }
    }
}